08_project/mohankrishna/pkg/mohankrishna-puppy/store/sync_store.go

```diff
@@ -0,0 +1,55 @@
+package store
+
+import (
+	"fmt"
+	"sync"
+
+	types "github.com/anz-bank/go-training/08_project/mohankrishna/pkg/mohankrishna-puppy"
+)
+
+type SyncStore struct {
+	sm sync.Map
+}
+
+func NewSyncStore() *SyncStore {
+	return &SyncStore{}
+}
+
+func (ss *SyncStore) CreatePuppy(puppy *types.Puppy) error {
```

Comment:
The multiple store accesses here create a race condition.
Add the following lines to the start of this function (with the classes mutex)
```go
mutex.Lock()
defer mutex.Unlock()
```